src/main/java/com/dell/pravega/perf/WriterWorker.java

```diff
@@ -0,0 +1,131 @@
+     *
+     * @param data   data to write
+     * @param record to call for benchmarking
+     * @return time return the data sent time
+     */
+    public abstract long recordWrite(String data, TriConsumer record);
+
+    /**
+     * writes the data and benchmark
+     *
+     * @param data data to write
+     */
+    public abstract void writeData(String data);
+
+
+    /**
+     * flush the producer data.
+     */
+    public abstract void flush();
+
+    @Override
+    public Void call() throws InterruptedException, ExecutionException, IOException {
+        perf.benchmark();
+        return null;
+    }
+
+    private class EventsWriter implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException, IOException {
+            // Construct event payload
+            String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+            String payload = String.format("%-" + messageSize + "s", val);
+
+            for (int i = 0; i < events; i++) {
+                recordWrite(payload, stats::recordTime);
+                eCnt.control(i);
+            }
+            flush();
+        }
+    }
+
+    private class EventsWriterRW implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException, IOException {
+            for (int i = 0; i < events; i++) {
+                // Construct event payload
+                String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+                String payload = String.format("%-" + messageSize + "s", val);
+                writeData(payload);
+                eCnt.control(i);
+            }
+
+            flush();
+
+        }
+    }
+
+
+    private class EventsWriterTime implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException, IOException {
+            // Construct event payload
+            String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+            String payload = String.format("%-" + messageSize + "s", val);
+            long time = System.currentTimeMillis();
+
+            for (int i = 0; ((time - StartTime) / 1000) < secondsToRun; i++) {
+                time = recordWrite(payload, stats::recordTime);
+                eCnt.control(i);
+            }
+
+            flush();
+        }
+    }
+
+    private class EventsWriterTimeRW implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException, IOException {
+            long time = System.currentTimeMillis();
+            for (int i = 0; ((time - StartTime) / 1000) < secondsToRun; i++) {
+                time = System.currentTimeMillis();
+                String val = time + ", " + workerID + ", " + (int) (Math.random() * 200);
+                String payload = String.format("%-" + messageSize + "s", val);
```

Comment:
Here and elsewhere, it doesn't make sense to pass a concatenated string into a string formater. It does concatenation automatically. 
This is going to end up creating 4 intermediate string objects on the line above and 3 more here, only to end up using a formater that could do it all at once.

Reply:
ok, I have changed to simple string concat .. now